pkg/apitypes/managed_tx.go

```diff
@@ -179,6 +181,85 @@ func (mtx *ManagedTX) SetSequence(i int64) {
 	mtx.SequenceID = fmt.Sprintf("%.12d", i)
 }
 
+// ApplyExternalTxUpdates applies the external updates to the managed transaction
+// and returns the updates that were applied
+// ApplyExternalTxUpdates applies the external updates to the managed transaction
+// and returns the updates that were applied, whether any update occurred, and a map
+// of changed fields with their old and new values for easy JSON marshalling.
+func (mtx *ManagedTX) ApplyExternalTxUpdates(txUpdate TXUpdatesExternal) (txUpdates TXUpdates, updated bool, valueChangeMap map[string]map[string]string) {
```

Comment:
So the design I believe is that this is a helper function that policy implementations (which hold the complex threading model) can call back into, at the point the "task" of editing a transaction has made it to the right locked in-memory code that is safe to make the change.

Will validate this understanding through the rest of the review.